src/tapeimage.cpp

```diff
@@ -419,23 +419,41 @@ void tapeimage::seek_with_index(std::int64_t n) noexcept (false) {
          return h.next < n + zero;
     };
 
-    auto cur = std::lower_bound(
+    auto lower = std::lower_bound(
         this->markers.begin(),
         this->markers.end(),
         n,
         less
     );
 
-    //phase 2
-    while(true) {
-        if (!this->search_further(cur, n)) {
-            break;
-        }
-        ++cur;
-    }
+    /*
+     * We found the right record when record->next > n. All hits after will
+     * also be a match, but this is ok since the search is in an ordered list.
+     *
+     * Using a mutable lambda to carry the header contribution is a pretty
+     * convoluted approach, but both the element *and* the header sizes need to
+     * be accounted for, and the latter is only available through the
+     * *position* in the index, which doesn't play so well with the std
+     * algorithms. The use of lambda + find-if is still valuable though, as it
+     * gives a clean error check if the offset n is somehow *not* in the index.
+     */
+    auto overhead = this->protocol_overhead(lower);
+    const auto next_larger = [this, n, overhead] (const header& rec) mutable {
+        const auto is_larger = n + overhead <= rec.next;
```

Comment:
OK, I was trying to abstain but failed :smile:  Again, feel free to ignore, but I consider my duty done :D 

This whole method is not straightforward and with confusing names it becomes even more confusing.
I am not the one to speak about names ( :smile: ), but `less`, `next_larger`, `is_larger` do not make it clear what is supposed to be lesser or larger than what. Just suggestions regarding what I mean:

next_larger -> n_in_larger_header (we hop further and further to find a header where n belongs)
is_larger ->  is_n_reached (to communicate that we have to stop, n is already behind)
less -> n_in_smaller_header (to show relation between n and expected header)